test/test-ParametersParser.cc

```diff
@@ -84,7 +84,7 @@ TEST_CASE("Parse illegal parameter among legal ones", "[parameters]")
     "+ Parameters\n"\
     "    + OK_1\n"\
     "    + i;legal\n"\
-    "    + OK-2\n";
+    "    + OK_2\n";
```

Comment:
do I understand correctly, it changes behavior for old syntax?
If yes, I mean we should add some kind of deprecation warning.